File: 정수빈/LeetCode/medium/simplify-path.py
# https://leetcode.com/problems/simplify-path/?envType=study-plan-v2&envId=top-interview-150

class Solution:
    def simplifyPath(self, path: str) -> str:
        path = collections.deque(path.split("/"))
        # print(path)
        s = [path.popleft()]
        while path:
            p = path.popleft()
            if p == "":
                pass
            elif p == "..":
                if s:
                    s.pop()
            elif p == ".":
                pass
            else:
                s.append(p)
            # print(s)
        path = ""
        for c in s:
            if c != "":
                path += "/"+c
        return path if len(path) > 0 else "/"

File: 장세훈/LeetCode/medium/SimplifyPath.java
class SimplifyPath {

  /*
      https://leetcode.com/problems/simplify-path/
   */
  public String simplifyPath(String path) {
    String[] split = path.split("/");
    Deque<String> queue = new ArrayDeque<>();

    for (String s : split) {
      if (s.isEmpty() || s.equals(".")) {
        continue;
      } else if (s.equals("..")) {
        if (!queue.isEmpty()) {
          queue.pollLast();
        }
      } else {
        queue.addLast(s);
      }
    }

    StringBuilder sb = new StringBuilder();
    while (!queue.isEmpty()) {
      sb.append("/").append(queue.pollFirst());
    }

    return sb.isEmpty() ? "/" : sb.toString();
  }

}